sympy/core/tests/test_numbers.py

```diff
@@ -436,6 +436,10 @@ def eq(a, b):
     a = Float(2) ** Float(4)
     assert eq(a.evalf(), Float(16))
     assert (S(.3) == S(.5)) is False
+
+    # https://github.com/sympy/sympy/issues/20796
+    assert (S(0.0) == S.false) is False
```

Comment:
The issue mentions 4 different combinations of comparison. Are there any other tests for the other combinations?

It would be better for this test to be with the other tests if there are any. If not then we should test all cases here or make a particular test function that can test lots of combinations. I think `Integer` (i.e. `S(0)`) should also be tested.

Reply:
I've checked `test_numbers` and `test_sympify`, but I can't seem to find any already existing tests for those others. So should I create a test function in `test_numbers` or somewhere else? And could you provide the specific test you were thinking of for `Integer`?

Reply:
For eaxmple:
```
In [1]: S(0) == S.false
Out[1]: False
```
We should make sure all cases are covered.

Reply:
Got it. And just put it in `test_numbers`?

Reply:
Yes.

Reply:
How's that?

Reply:
LGTM

Reply:
Great.